lib/config/optionsconfiguration.go

```diff
@@ -43,6 +43,7 @@ type OptionsConfiguration struct {
 	TempIndexMinBlocks      int      `xml:"tempIndexMinBlocks" json:"tempIndexMinBlocks" default:"10"`
 	UnackedNotificationIDs  []string `xml:"unackedNotificationID" json:"unackedNotificationIDs"`
 	TrafficClass            int      `xml:"trafficClass" json:"trafficClass"`
+	WeakHashSelectionMethod int      `xml:"weakHashSelectionMethod" json:"weakHashSelectionMethod"`
```

Comment:
Magic numbers are magic. This should be an enum type with constants that serialize to a string in the config. And probably something like `WeakHashEnabled` with possible values `auto`, `yes`, `no` (yes and no being called something else if you prefer)? Yes there's some overlap with the weak hash threshold, but it would then be back to defaulting to 25 and it was never very intuitive to have to set a percentage to >100 to disable something.